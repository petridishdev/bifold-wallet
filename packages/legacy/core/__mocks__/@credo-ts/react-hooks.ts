/* eslint-disable @typescript-eslint/no-explicit-any */
import { LegacyIndyCredentialFormat } from '@credo-ts/anoncreds'
import {
  BasicMessageRecord,
  CredentialExchangeRecord,
  CredentialProtocolOptions,
  ProofExchangeRecord,
  ConnectionRecord,
  DidExchangeRole,
  DidExchangeState,
} from '@credo-ts/core'

const useCredentials = jest.fn().mockReturnValue({ records: [] } as any)
const useProofs = jest.fn().mockReturnValue({ records: [] } as any)
const useCredentialByState = jest.fn().mockReturnValue([] as CredentialExchangeRecord[])
const useProofByState = jest.fn().mockReturnValue([] as ProofExchangeRecord[])
const useBasicMessagesByConnectionId = jest.fn().mockReturnValue([] as BasicMessageRecord[])
const useBasicMessages = jest.fn().mockReturnValue({ records: [] as BasicMessageRecord[] })
const mockCredentialModule = {
  acceptOffer: jest.fn(),
  declineOffer: jest.fn(),
  getFormatData: jest
    .fn()
    .mockReturnValue(
      Promise.resolve({} as CredentialProtocolOptions.GetCredentialFormatDataReturn<[LegacyIndyCredentialFormat]>)
    ),
  findAllByQuery: jest.fn().mockReturnValue(Promise.resolve([])),
}
const mockProofModule = {
  getCredentialsForRequest: jest.fn(),
  acceptRequest: jest.fn(),
  declineRequest: jest.fn(),
  getFormatData: jest.fn(),
  findRequestMessage: jest.fn(),
  requestProof: jest.fn(),
  update: jest.fn(),
  findAllByQuery: jest.fn().mockReturnValue(Promise.resolve([])),
}
const mockBasicMessagesModule = {
  findAllByQuery: jest.fn().mockReturnValue(Promise.resolve([])),
}
const mockConnectionsModule = {
  getAll: jest.fn().mockReturnValue(
    Promise.resolve([
      new ConnectionRecord({
        id: '1',
        did: '9gtPKWtaUKxJir5YG2VPxX',
        theirLabel: 'Faber',
        role: DidExchangeRole.Responder,
        theirDid: '2SBuq9fpLT8qUiQKr2RgBe',
        threadId: '1',
        state: DidExchangeState.Completed,
        createdAt: new Date('2020-01-02T00:00:00.000Z'),
      }),
      new ConnectionRecord({
        id: '2',
        did: '2SBuq9fpLT8qUiQKr2RgBe',
        role: DidExchangeRole.Requester,
        theirLabel: 'Bob',
        theirDid: '9gtPKWtaUKxJir5YG2VPxX',
        threadId: '1',
        state: DidExchangeState.Completed,
        createdAt: new Date('2020-01-01T00:00:00.000Z'),
      }),
    ])
  ),
}

const mockMediationRecipient = {
  initiateMessagePickup: jest.fn(),
}
const mockOobModule = {
  findById: jest.fn().mockReturnValue(Promise.resolve(null)),
  createInvitation: jest.fn(),
  toUrl: jest.fn(),
}
const mockBasicMessageRepository = {
  update: jest.fn(),
}
const mockAgentContext = {
  dependencyManager: {
    resolve: jest.fn().mockReturnValue(mockBasicMessageRepository),
  },
}
const useAgent = () => ({
  agent: {
    credentials: mockCredentialModule,
    proofs: mockProofModule,
    basicMessages: mockBasicMessagesModule,
    connections: mockConnectionsModule,
    mediationRecipient: mockMediationRecipient,
    oob: mockOobModule,
    context: mockAgentContext,
    receiveMessage: jest.fn(),
  },
})
const useCredentialById = jest.fn()
const useProofById = jest.fn()
const useConnectionById = jest.fn()
const useConnections = jest.fn()

export {
  useAgent,
  useBasicMessages,
  useBasicMessagesByConnectionId,
  useConnectionById,
  useConnections,
  useCredentialById,
  useCredentialByState,
  useCredentials,
  useProofById,
  useProofByState,
  useProofs,
}
