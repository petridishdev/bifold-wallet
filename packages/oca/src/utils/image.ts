import { CaptureBaseAttributeType } from '@oca/types'
import { ImageSourcePropType } from 'react-native'

export const toImageSource = (source: unknown): ImageSourcePropType => {
  if (typeof source === 'string') {
    return { uri: source as string }
  }
  return source as ImageSourcePropType
}

export const isDataUrl = (value: string | number | null) => {
  return typeof value === 'string' && value.startsWith('data:image/')
}

export const isBinaryType = (type: string | undefined) => {
  return CaptureBaseAttributeType.Binary === type
}
