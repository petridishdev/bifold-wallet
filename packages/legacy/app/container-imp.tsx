import { BaseLogger } from '@credo-ts/core'
import { Container, TokenMapping } from '@hyperledger/aries-bifold-core'
import { DependencyContainer } from 'tsyringe'
export class AppContainer implements Container {
  private _container: DependencyContainer
  private log?: BaseLogger

  public constructor(bifoldContainer: Container, log?: BaseLogger) {
    this._container = bifoldContainer.container.createChildContainer()
    this.log = log
  }

  public get container(): DependencyContainer {
    return this._container
  }

  public init(): Container {
    // eslint-disable-next-line no-console
    this.log?.info(`Initializing App container`)
    // Here you can register any component to override components in core package
    // Example: Replacing button in core with custom button
    // this.container.registerInstance(TOKENS.COMP_BUTTON, Button)
    return this
  }

  public resolve<K extends keyof TokenMapping>(token: K): TokenMapping[K] {
    return this._container.resolve(token) as TokenMapping[K]
  }
  public resolveAll<K extends keyof TokenMapping, T extends K[]>(
    tokens: [...T]
  ): { [I in keyof T]: TokenMapping[T[I]] } {
    return tokens.map((key) => this.resolve(key)!) as { [I in keyof T]: TokenMapping[T[I]] }
  }
}
