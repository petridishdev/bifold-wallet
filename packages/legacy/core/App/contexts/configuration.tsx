import { Agent } from '@credo-ts/core'
import { ProofRequestTemplate } from '@hyperledger/aries-bifold-verifier'
import { StackNavigationOptions, StackScreenProps } from '@react-navigation/stack'
import { ReducerAction, createContext, useContext } from 'react'

import { EmptyListProps } from '../components/misc/EmptyList'
import { RecordProps } from '../components/record/Record'
import { Locales } from '../localization'
import OnboardingPages from '../screens/OnboardingPages'
import { ConnectStackParams } from '../types/navigators'
import { PINSecurityParams } from '../types/security'
import { SettingSection } from '../types/settings'

interface NotificationConfiguration {
  component: React.FC
  onCloseAction: (dispatch?: React.Dispatch<ReducerAction<any>>) => void
  title: string
  description: string
  buttonTitle: string
  pageTitle: string
}

interface PushNotificationConfiguration {
  // function to get the current push notification permission status
  status: () => Promise<'denied' | 'granted' | 'unknown'>
  // function to request permission for push notifications
  setup: () => Promise<'denied' | 'granted' | 'unknown'>
  //function to call when the user changes the push notification setting
  toggle: (state: boolean, agent: Agent<any>) => Promise<void>
}

export interface ConfigurationContext {
  pages: typeof OnboardingPages
  splash: React.FC
  terms: React.FC
  preface: React.FC
  homeHeaderView: React.FC
  homeFooterView: React.FC
  credentialListHeaderRight: React.FC
  credentialListOptions: React.FC
  credentialEmptyList: React.FC<EmptyListProps>
  developer: React.FC
  proofTemplateBaseUrl?: string
  scan: React.FC<StackScreenProps<ConnectStackParams>>
  record: React.FC<RecordProps>
  PINSecurity: PINSecurityParams
  settings: SettingSection[]
  customNotification: NotificationConfiguration
  supportedLanguages: Locales[]
  connectionTimerDelay?: number
  autoRedirectConnectionToHome?: boolean
  proofRequestTemplates?: (useDevTemplates: boolean) => Array<ProofRequestTemplate>
  enableTours?: boolean
  enableImplicitInvitations?: boolean
  enableReuseConnections?: boolean
  showPreface?: boolean
  disableOnboardingSkip?: boolean
  useBiometry: React.FC
  enablePushNotifications?: PushNotificationConfiguration
  useCustomNotifications: () => { total: number; notifications: any }
  useAttestation?: () => { start: () => void; stop: () => void; loading: boolean }
  whereToUseWalletUrl: string
  showScanHelp?: boolean
  showScanButton?: boolean
  globalScreenOptions?: StackNavigationOptions
  showDetailsInfo?: boolean
  contactHideList?: string[]
  credentialHideList?: string[]
}

export const ConfigurationContext = createContext<ConfigurationContext>(null as unknown as ConfigurationContext)

export const ConfigurationProvider = ConfigurationContext.Provider

export const useConfiguration = () => useContext(ConfigurationContext)
