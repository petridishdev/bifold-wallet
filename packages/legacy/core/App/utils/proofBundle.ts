import { BaseLogger } from '@credo-ts/core'
import {
  AnonCredsProofRequestTemplatePayload,
  ProofRequestTemplate,
  useProofRequestTemplates,
} from '@hyperledger/aries-bifold-verifier'
import axios, { AxiosError } from 'axios'

import { useConfiguration } from '../contexts/configuration'

const calculatePreviousYear = (yearOffset: number) => {
  const pastDate = new Date()
  pastDate.setFullYear(pastDate.getFullYear() + yearOffset)
  return parseInt(pastDate.toISOString().split('T')[0].replace(/-/g, ''))
}

export const applyTemplateMarkers = (templates: any): any => {
  if (!templates) return templates
  const markerActions: { [key: string]: (param: string) => string } = {
    now: () => Math.floor(new Date().getTime() / 1000).toString(),
    currentDate: (offset: string) => calculatePreviousYear(parseInt(offset)).toString(),
  }
  let templateString = JSON.stringify(templates)
  // regex to find all markers in the template so we can replace them with computed values
  const markers = [...templateString.matchAll(/"@\{(\w+)(?:\((\S*)\))?\}"/gm)]

  markers.forEach((marker) => {
    const markerValue = markerActions[marker[1] as string](marker[2])
    templateString = templateString.replace(marker[0], markerValue)
  })
  return JSON.parse(templateString)
}

export const applyDevRestrictions = (templates: ProofRequestTemplate[]): ProofRequestTemplate[] => {
  return templates.map((temp) => {
    return {
      ...temp,
      payload: {
        ...temp.payload,
        data: (temp.payload as AnonCredsProofRequestTemplatePayload).data.map((data) => {
          return {
            ...data,
            requestedAttributes: data.requestedAttributes?.map((attr) => {
              return {
                ...attr,
                restrictions: [...(attr.restrictions ?? []), ...(attr.devRestrictions ?? [])],
                devRestrictions: [],
              }
            }),
            requestedPredicates: data.requestedPredicates?.map((pred) => {
              return {
                ...pred,
                restrictions: [...(pred.restrictions ?? []), ...(pred.devRestrictions ?? [])],
                devRestrictions: [],
              }
            }),
          }
        }),
      },
    }
  })
}

export interface ProofBundleResolverType {
  resolve: (acceptDevRestrictions: boolean) => Promise<ProofRequestTemplate[] | undefined>
  resolveById: (templateId: string, acceptDevRestrictions: boolean) => Promise<ProofRequestTemplate | undefined>
}

export const useRemoteProofBundleResolver = (
  indexFileBaseUrl: string | undefined,
  log?: BaseLogger
): ProofBundleResolverType => {
  if (indexFileBaseUrl) {
    return new RemoteProofBundleResolver(indexFileBaseUrl, log)
  } else {
    return new DefaultProofBundleResolver()
  }
}

export class RemoteProofBundleResolver implements ProofBundleResolverType {
  private remoteServer
  private templateData: ProofRequestTemplate[] | undefined
  private log?: BaseLogger

  public constructor(indexFileBaseUrl: string, log?: BaseLogger) {
    this.remoteServer = axios.create({
      baseURL: indexFileBaseUrl,
    })
    this.log = log
  }

  public async resolve(acceptDevRestrictions: boolean): Promise<ProofRequestTemplate[] | undefined> {
    if (this.templateData) {
      let templateData = this.templateData

      if (acceptDevRestrictions) {
        templateData = applyDevRestrictions(templateData)
      }

      return Promise.resolve(templateData)
    }

    return this.remoteServer
      .get('proof-templates.json')
      .then((response) => {
        this.log?.info('Fetched proof templates')

        try {
          let templateData: ProofRequestTemplate[] = response.data
          this.templateData = templateData

          if (acceptDevRestrictions) {
            templateData = applyDevRestrictions(templateData)
          }

          return templateData
        } catch (error: unknown) {
          this.log?.error('Failed to parse proof templates', error as Error)

          return undefined
        }
      })
      .catch((error: AxiosError) => {
        this.log?.error('Failed to fetch proof templates', error)

        return undefined
      })
  }

  public async resolveById(
    templateId: string,
    acceptDevRestrictions: boolean
  ): Promise<ProofRequestTemplate | undefined> {
    if (!this.templateData) {
      return (await this.resolve(acceptDevRestrictions))?.find((template) => template.id === templateId)
    }

    let templateData = this.templateData

    if (acceptDevRestrictions) {
      templateData = applyDevRestrictions(templateData)
    }
    const template = templateData.find((template) => template.id === templateId)

    return template
  }
}

export class DefaultProofBundleResolver implements ProofBundleResolverType {
  private proofRequestTemplates

  public constructor() {
    const { proofRequestTemplates } = useConfiguration()
    this.proofRequestTemplates = proofRequestTemplates ?? useProofRequestTemplates
  }

  public async resolve(acceptDevRestrictions: boolean): Promise<ProofRequestTemplate[]> {
    return Promise.resolve(this.proofRequestTemplates(acceptDevRestrictions))
  }

  public async resolveById(
    templateId: string,
    acceptDevRestrictions: boolean
  ): Promise<ProofRequestTemplate | undefined> {
    return Promise.resolve(
      this.proofRequestTemplates(acceptDevRestrictions).find((template) => template.id === templateId)
    )
  }
}
