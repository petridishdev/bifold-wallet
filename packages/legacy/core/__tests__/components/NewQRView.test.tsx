import { useAgent, useConnections } from '@credo-ts/react-hooks'
import { act, render } from '@testing-library/react-native'
import React from 'react'

import NewQRView from '../../App/components/misc/NewQRView'
import { StoreProvider, defaultState } from '../../App/contexts/store'
import { testIdWithKey } from '../../App/utils/testable'
import { useNavigation } from '@react-navigation/native'
import { BasicAppContext } from '../helpers/app'

jest.mock('react-native-orientation-locker', () => {
  return require('../../__mocks__/custom/react-native-orientation-locker')
})


describe('NewQRView Component', () => {
  beforeAll(() => {
    jest.useFakeTimers()
  })

  afterAll(() => {
    jest.useRealTimers()
  })

  beforeEach(() => {
    jest.clearAllMocks()
    // @ts-ignore
    useConnections.mockReturnValue({ records: [] })
  })

  const navigation = useNavigation()

  test('Renders correctly on first tab', async () => {
    const tree = render(
      <BasicAppContext>
        <NewQRView
          defaultToConnect={false}
          handleCodeScan={() => Promise.resolve()}
          navigation={navigation as any}
          route={{} as any}
        />
      </BasicAppContext>
    )

    await act(() => {
      jest.runAllTimers()
    })

    expect(tree).toMatchSnapshot()
  })

  test('Renders correctly on second tab', async () => {
    // @ts-ignore
    useAgent().agent?.oob.createInvitation.mockReturnValue({
      outOfBandInvitation: {
        toUrl: () => {
          return ''
        },
      },
    })
    const tree = render(
      <StoreProvider
        initialState={{
          ...defaultState,
          preferences: {
            ...defaultState.preferences,
            walletName: 'My Wallet - 1234',
          },
        }}
      >
        <BasicAppContext>
          <NewQRView
            defaultToConnect={true}
            handleCodeScan={() => Promise.resolve()}
            navigation={navigation as any}
            route={{} as any}
          />
        </BasicAppContext>
      </StoreProvider>
    )

    await act(() => {
      jest.runAllTimers()
    })

    expect(tree).toMatchSnapshot()
  })

  test('Contains test IDs', async () => {
    // @ts-ignore
    useAgent().agent?.oob.createInvitation.mockReturnValue({
      outOfBandInvitation: {
        toUrl: () => {
          return ''
        },
      },
    })
    const { getByTestId } = render(
      <StoreProvider
        initialState={{
          ...defaultState,
          preferences: {
            ...defaultState.preferences,
            walletName: 'My Wallet - 1234',
          },
        }}
      >
        <BasicAppContext>
          <NewQRView
            defaultToConnect={true}
            handleCodeScan={() => Promise.resolve()}
            navigation={navigation as any}
            route={{} as any}
          />
        </BasicAppContext>
      </StoreProvider>
    )

    await act(() => {
      jest.runAllTimers()
    })

    const editButtonByTestId = getByTestId(testIdWithKey('EditWalletName'))

    expect(editButtonByTestId).not.toBeNull()
  })
})
