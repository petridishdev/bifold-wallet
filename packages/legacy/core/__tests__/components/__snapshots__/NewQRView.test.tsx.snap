// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewQRView Component Renders correctly on first tab 1`] = `
Array [
  <RNCSafeAreaView
    edges={
      Array [
        "left",
        "right",
      ]
    }
    style={
      Object {
        "backgroundColor": "#313132",
        "flex": 1,
      }
    }
  >
    <Modal
      animationType="fade"
      hardwareAccelerated={false}
      transparent={true}
      visible={false}
    />
    <View
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "transform": Array [
              Object {
                "rotate": "0deg",
              },
            ],
          },
        ]
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "flexDirection": "row",
            "marginHorizontal": 40,
            "paddingTop": 30,
          }
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 40,
              },
              Object {
                "color": "#FFFFFF",
                "padding": 4,
              },
              Object {
                "fontFamily": "Material Design Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          󰐳
        </Text>
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 20,
              "fontWeight": "bold",
              "marginHorizontal": 10,
              "textAlign": "center",
            }
          }
        >
          Scan.WillScanAutomatically
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexGrow": 1,
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            Object {
              "borderColor": "#FFFFFF",
              "borderRadius": 24,
              "borderWidth": 2,
              "height": 250,
              "width": 250,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <View
          accessibilityLabel="Scan.ScanNow"
          accessibilityRole="button"
          accessibilityState={
            Object {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            Object {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          hitSlop={
            Object {
              "bottom": 44,
              "left": 44,
              "right": 44,
              "top": 44,
            }
          }
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "opacity": 1,
              },
            ]
          }
          testID="com.ariesbifold:id/ScanNow"
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 60,
                },
                Object {
                  "color": "white",
                  "marginBottom": -15,
                },
                Object {
                  "fontFamily": "Material Design Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            󰝦
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "height": 24,
            "marginBottom": 60,
            "marginHorizontal": 24,
            "width": "90%",
          }
        }
      >
        <View
          accessibilityLabel="Scan.ScanHelp"
          accessibilityRole="button"
          accessibilityState={
            Object {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            Object {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          hitSlop={
            Object {
              "bottom": 44,
              "left": 44,
              "right": 44,
              "top": 44,
            }
          }
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "opacity": 1,
              },
            ]
          }
          testID="com.ariesbifold:id/ScanHelp"
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 24,
                },
                Object {
                  "color": "white",
                },
                Object {
                  "fontFamily": "Material Design Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            󰋗
          </Text>
        </View>
        <View
          style={
            Object {
              "marginLeft": "auto",
              "width": 10,
            }
          }
        />
        <View
          accessibilityLabel="Scan.Torch"
          accessibilityRole="button"
          accessibilityState={
            Object {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            Object {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          hitSlop={
            Object {
              "bottom": 44,
              "left": 44,
              "right": 44,
              "top": 44,
            }
          }
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": undefined,
              "borderColor": "#FFFFFF",
              "borderRadius": 24,
              "borderWidth": 1,
              "height": 24,
              "justifyContent": "center",
              "marginBottom": 50,
              "opacity": 1,
              "width": 24,
            }
          }
          testID="com.ariesbifold:id/ScanTorch"
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 24,
                },
                Object {
                  "alignItems": "center",
                },
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#313132",
          "borderTopWidth": 0,
          "flexDirection": "row",
          "height": 60,
          "paddingBottom": 0,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": -3,
            "width": 0,
          },
          "shadowOpacity": 0.1,
          "shadowRadius": 6,
        }
      }
    >
      <View
        accessibilityLabel="Scan.ScanQRCode"
        accessibilityState={
          Object {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          Object {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
            "opacity": 1,
          }
        }
        testID="com.ariesbifold:id/Scan.ScanQRCode"
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#42803E",
                "fontSize": 30,
              },
              undefined,
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
      <View
        accessibilityLabel="Scan.MyQRCode"
        accessibilityState={
          Object {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          Object {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
            "opacity": 1,
          }
        }
        testID="com.ariesbifold:id/Scan.MyQRCode"
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 30,
              },
              undefined,
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </RNCSafeAreaView>,
  <RNCSafeAreaView
    edges={
      Array [
        "bottom",
      ]
    }
    style={
      Object {
        "backgroundColor": "#000000",
        "flex": 0,
      }
    }
  />,
]
`;

exports[`NewQRView Component Renders correctly on second tab 1`] = `
Array [
  <RNCSafeAreaView
    edges={
      Array [
        "left",
        "right",
      ]
    }
    style={
      Object {
        "backgroundColor": "#313132",
        "flex": 1,
      }
    }
  >
    <RCTScrollView>
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "marginTop": 10,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                }
              }
              testID="com.ariesbifold:id/LoadingActivityIndicator"
            >
              <Image
                source={
                  Object {
                    "default": "",
                  }
                }
                style={
                  Object {
                    "height": "33%",
                    "width": "33%",
                  }
                }
                testID="com.ariesbifold:id/LoadingActivityIndicatorImage"
              />
              <View
                collapsable={false}
                style={
                  Object {
                    "position": "absolute",
                    "transform": Array [
                      Object {
                        "rotate": "360deg",
                      },
                    ],
                  }
                }
              >
                <
                  fill="#FFFFFF"
                  height={200}
                  width={200}
                />
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "flex": 1,
                "paddingHorizontal": 20,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#FFFFFF",
                      "fontSize": 32,
                      "fontWeight": "bold",
                      "textAlign": "center",
                    },
                    Object {
                      "paddingHorizontal": 20,
                    },
                  ]
                }
                testID="com.ariesbifold:id/WalletName"
              >
                My Wallet - 1234
              </Text>
              <View
                accessibilityLabel="NameWallet.EditWalletName"
                accessibilityRole="button"
                accessibilityState={
                  Object {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  Object {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                hitSlop={
                  Object {
                    "bottom": 44,
                    "left": 44,
                    "right": 44,
                    "top": 44,
                  }
                }
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                testID="com.ariesbifold:id/EditWalletName"
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "marginLeft": 0,
                      "marginRight": 15,
                      "minHeight": 26,
                      "minWidth": 26,
                    }
                  }
                >
                  <Text
                    allowFontScaling={false}
                    selectable={false}
                    style={
                      Array [
                        Object {
                          "color": "#FFFFFF",
                          "fontSize": 26,
                        },
                        undefined,
                        Object {
                          "fontFamily": "Material Design Icons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                  >
                    󰏫
                  </Text>
                </View>
              </View>
            </View>
            <Text
              style={
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 18,
                  "fontWeight": "normal",
                  "marginTop": 20,
                  "textAlign": "center",
                }
              }
            >
              Connection.ShareQR
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
    <View
      style={
        Object {
          "backgroundColor": "#313132",
          "borderTopWidth": 0,
          "flexDirection": "row",
          "height": 60,
          "paddingBottom": 0,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": -3,
            "width": 0,
          },
          "shadowOpacity": 0.1,
          "shadowRadius": 6,
        }
      }
    >
      <View
        accessibilityLabel="Scan.ScanQRCode"
        accessibilityState={
          Object {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          Object {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
            "opacity": 1,
          }
        }
        testID="com.ariesbifold:id/Scan.ScanQRCode"
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 30,
              },
              undefined,
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
      <View
        accessibilityLabel="Scan.MyQRCode"
        accessibilityState={
          Object {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          Object {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
            "opacity": 1,
          }
        }
        testID="com.ariesbifold:id/Scan.MyQRCode"
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#42803E",
                "fontSize": 30,
              },
              undefined,
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </RNCSafeAreaView>,
  <RNCSafeAreaView
    edges={
      Array [
        "bottom",
      ]
    }
    style={
      Object {
        "backgroundColor": "#000000",
        "flex": 0,
      }
    }
  />,
]
`;
